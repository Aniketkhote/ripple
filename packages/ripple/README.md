<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://raw.githubusercontent.com/trueadm/ripple/master/assets/ripple-dark.png">
  <img src="https://raw.githubusercontent.com/trueadm/ripple/master/assets/ripple-light.png" alt="Ripple - the elegant UI framework for the web" />
</picture>

# What is Ripple?

> Currently, this project is still in early development, and should not be used in production.

Ripple is a TypeScript UI framework for the web. To find out more, view [Ripple's Github README](https://github.com/trueadm/ripple).
