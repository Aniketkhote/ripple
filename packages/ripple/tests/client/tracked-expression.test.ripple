import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { mount, flushSync, TrackedSet, track } from 'ripple';

describe('TrackedExpression tests', () => {
	let container;

	function render(component) {
		mount(component, {
			target: container
		});
	}

	beforeEach(() => {
		container = document.createElement('div');
		document.body.appendChild(container);
	});

	afterEach(() => {
		document.body.removeChild(container);
		container = null;
	});

	it('should handle the syntax correctly', () => {
		component App() {
      let count = track(0);

      function get_count() {
        return count;
      }


      <div>{@(count)}</div>
      <div>{@(get_count())}</div>
      <div>{++@(count)}</div>
      <div>{++@(get_count())}</div>
      <div>{@(count)++}</div>
      <div>{@(get_count())++}</div>
      <div>{@(count)}</div>
      <div>{!@(count)}</div>
      <div>{!!@(count)}</div>
    }

    render(App);
    expect(container).toMatchSnapshot();
	});
});