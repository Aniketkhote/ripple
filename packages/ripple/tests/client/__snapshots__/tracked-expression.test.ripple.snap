// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TrackedExpression tests > should handle the syntax correctly 1`] = `
<div>
  <div>
    0
  </div>
  <div>
    0
  </div>
  <div>
    1
  </div>
  <div>
    2
  </div>
  <div>
    2
  </div>
  <div>
    3
  </div>
  <div>
    4
  </div>
  <div>
    false
  </div>
  <div>
    true
  </div>
  
</div>
`;
